tendrl/ceph_integration/objects/pool/atoms/check_pool_available/__init__.py

```diff
@@ -0,0 +1,43 @@
+import gevent
+
+from tendrl.ceph_integration.objects.pool import Pool
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons import objects
+from tendrl.commons.objects import AtomExecutionFailedError
+
+
+class CheckPoolAvailable(objects.BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(CheckPoolAvailable, self).__init__(*args, **kwargs)
+
+    def run(self):
+        retry_count = 0
+        while True:
+            pools = NS._int.client.read(
+                "clusters/%s/Pools" % NS.tendrl_context.integration_id
+            )
+            for entry in pools.leaves:
+                pool = Pool(pool_id=entry.key.split("Pools/")[-1]).load()
+                if pool.pool_name == self.parameters['Pool.poolname']:
+                    return True
+            retry_count += 1
+            gevent.sleep(1)
```

Comment:
what if the sds sync interval becomes configurable and is set to 50 seconds

This will time out in that case, Id suggest you dont have any timeout since we already know the Pool or object create atom was successful since we have reached the IsAvailable atom